cpp/betweenness_centrality_module/algorithm_online/betweenness_centrality_online.hpp

```diff
@@ -0,0 +1,255 @@
+template <typename T>
+void RemoveDuplicates(std::vector<T> &vector);
+
+namespace online_bc {
+enum class Operation { CREATE_EDGE, CREATE_NODE, CREATE_ATTACH_NODE, DELETE_EDGE, DELETE_NODE, DETACH_DELETE_NODE };
+
+std::unordered_set<std::uint64_t> NeighborsMemgraphIDs(const mg_graph::GraphView<> &graph,
+                                                       const std::vector<mg_graph::Neighbour<uint64_t>> neighbors);
+
+class OnlineBC {
+ private:
+  /// Maps nodes (Memgraph IDs) to their betweenness centrality scores
+  std::unordered_map<std::uint64_t, double> node_bc_scores;
+
+  /// Betweenness centrality score initialization flag
+  bool initialized = false;
+
+  /// Avoids counting edges twice in case of undirected graphs
+  static constexpr double NO_DOUBLE_COUNT = 2.0;
+
+  ///@brief Returns whether previously calculated betweenness centrality scores are inconsistent with the current graph,
+  /// i.e. if the nodes which have associated betweenness centrality scores are not the same as the graph’s nodes.
+  ///
+  ///@param graph Current graph
+  bool Inconsistent(const mg_graph::GraphView<> &graph) const;
+
+  ///@brief Normalize each node’s betweenness centrality score by the number of node pairs not
+  /// containing said node. The normalization constant is 2/((N-1)(N-2)) for undirected and 1/((N-1)(N-2)) for directed
+  /// graphs, with N being the number of graph nodes.
+  ///
+  ///@param node_bc_scores Graph nodes with associated betweenness centrality scores
+  ///@param graph_order Order (Nº of nodes) of the graph
+  ///
+  ///@return Normalized betweenness centrality scores
+  std::unordered_map<std::uint64_t, double> NormalizeBC(const std::unordered_map<std::uint64_t, double> &node_bc_scores,
+                                                        const std::uint64_t graph_order) const;
+
+  ///@brief Wrapper for the offline betweenness centrality algorithm that maps betweenness centrality scores to their
+  /// nodes’ Memgraph IDs.
+  ///
+  ///@param graph Current graph
+  ///@param threads Number of concurrent threads
+  void CallBrandesAlgorithm(const mg_graph::GraphView<> &graph, const std::uint64_t threads);
+
+  ///@brief Performs a breadth-first search from a given node with the Brandes’ algorithm.
+  ///
+  ///@param graph Graph traversed the BFS
+  ///@param start_id Start node ID
+  ///@param compensate_for_deleted_node If true, guarantees correct results for the DETACH_DELETE_NODE operation
+  ///
+  ///@return Nº of shortest paths to each node from the start node
+  /// Each node’s immediate predecessors on the shortest paths from the start node
+  /// IDs of nodes visited by the BFS, in reverse order
+  std::tuple<std::unordered_map<std::uint64_t, std::uint64_t>,
+             std::unordered_map<std::uint64_t, std::set<std::uint64_t>>, std::vector<std::uint64_t>>
+  BrandesBFS(const mg_graph::GraphView<> &graph, const std::uint64_t start_id,
+             const bool compensate_for_deleted_node = false) const;
```

Comment:
Where exactly do we use the information from the Brandes' algorithm

Reply:
It is used in the cases handled by `NodeEdgeUpdate`, i.e. those where a single iteration of Brandes is enough to update BC values.